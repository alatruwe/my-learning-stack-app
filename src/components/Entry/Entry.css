.vertical-border {
  display: inline-block;
  font-size: calc(5vw + 16px);
  line-height: 0.9;
  margin: 0 0.2em;
  width: 1.5vw;
  border-left: 0.2em solid rgba(0, 0, 0, 0);
  background: linear-gradient(#fff, #fff) padding-box,
    repeating-linear-gradient(
        -65deg,
        #f8f9fa 12.5vw,
        #e9ecef 0 25vw,
        #dee2e6 0 37.5vw,
        #ced4da 0 50vw,
        #adb5bd 0 62.5vw,
        #6c757d 0 75vw,
        #495057 0 87.5vw,
        #343a40 0 100vw
      )
      border-box;
  background-size: 200vw 100vh;
  background-attachment: fixed;
}

.entry {
  display: flex;
  margin-bottom: 10%;
}
.vertical-border {
  display: flex;
  flex-direction: column;
}

.entry-info {
  display: flex;
  flex-direction: column;
  width: 100%;
}

.entry-date-mood,
.entry-tech,
.entry-notes {
  display: flex;
}

.entry-date-mood,
.entry-notes {
  justify-content: space-evenly;
}
.entry-tech {
  justify-content: center;
}
