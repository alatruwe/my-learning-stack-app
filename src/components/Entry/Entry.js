import React from "react";
import "./Entry.css";

class Entry extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showLearningNotes: false,
      showStrugglingNotes: false,
      deleted: false,
    };
  }
  static defaultProps = {
    history: {
      push: () => {},
    },
  };

  resetState = () => {
    this.setState({ showLearningNotes: false, showStrugglingNotes: false });
  };
  showLearningNotes = () => {
    this.setState({ showLearningNotes: true, showStrugglingNotes: false });
  };
  showStrugglingNotes = () => {
    this.setState({ showLearningNotes: false, showStrugglingNotes: true });
  };

  // API call to delete data
  handleDeleteEntry = (event) => {
    event.preventDefault();
  };

  renderLearningNotes = () => {
    return (
      <div>
        {this.props.learningNotes}
        <button className="btn" type="button" onClick={this.handleDeleteEntry}>
          Delete
        </button>
      </div>
    );
  };
  renderStrugglingNotes = () => {
    return (
      <div>
        {this.props.strugglingNotes}
        <button className="btn" type="button" onClick={this.handleDeleteEntry}>
          Delete
        </button>
      </div>
    );
  };

  render() {
    return (
      <section className="entry">
        <div className="vertical-border"></div>
        <div className="entry-info">
          <div onClick={this.resetState}>
            <div className="entry-date-mood">
              <p>{this.props.date}</p>
              <p>{/*this.renderMood(this.props.mood)*/}:-)</p>
            </div>
            <div className="entry-tech">
              <p>{this.props.tech}</p>{" "}
            </div>
          </div>

          <div className="entry-buttons">
            <button type="button" onClick={this.showLearningNotes}>
              I learned
            </button>
            <button type="button" onClick={this.showStrugglingNotes}>
              I struggled
            </button>
          </div>
          <div className="entry-notes">
            {this.state.showLearningNotes ? this.renderLearningNotes() : <></>}
            {this.state.showStrugglingNotes ? (
              this.renderStrugglingNotes()
            ) : (
              <></>
            )}
          </div>
          {this.state.deleted && <p>Deleted!</p>}
        </div>
      </section>
    );
  }
}

export default Entry;
