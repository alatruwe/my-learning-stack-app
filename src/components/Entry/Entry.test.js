import React from "react";
import Entry from "./Entry";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

describe(`Entry component`, () => {
  it("renders the component by default", () => {
    const wrapper = shallow(<Entry />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
