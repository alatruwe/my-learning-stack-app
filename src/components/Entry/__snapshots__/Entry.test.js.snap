// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry component renders the component by default 1`] = `
<section
  className="entry"
>
  <div
    className="vertical-border"
  />
  <div
    className="entry-info"
  >
    <div
      onClick={[Function]}
    >
      <div
        className="entry-date-mood"
      >
        <div />
        <div />
      </div>
      <div
        className="entry-tech"
      >
        <p />
         
      </div>
    </div>
    <div
      className="entry-buttons"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        I learned
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        I struggled
      </button>
    </div>
    <div
      className="entry-notes"
    />
  </div>
</section>
`;
